testing/regress/ecl/prefetch2.ecl

```diff
@@ -40,12 +40,12 @@ END;
 
 outrec trans1(in L) := TRANSFORM
   SELF.word := L.word;
-  SELF.doc := SORTED(Files.getSearchSuperIndex()(KEYED(kind = TS.kindType.TextEntry),keyed(word = l.word)), doc)[1].doc;
+  SELF.doc := SORT(Files.getSearchSuperIndex()(KEYED(kind = TS.kindType.TextEntry),keyed(word = l.word)), doc)[1].doc;
```

Comment:
This is reading a result from a superkey of two indexes.  The original result is showing a match from the first of the subfiles in the superkey.  I suspect Thor was hitting a match from the second index which occurred on an earlier node.
So
i) the original SORTED() is wrong
ii) Is it well defined which order matches come in from subindexes of a superindex?

@richardkchapman what is your opinion?

Reply:
I think subindexes are supposed to interleave - i.e. you should see matches from part 1 of subindex 1 then from part 1 of subindex 2 then from part 2 of subindex 1

Reply:
> I suspect Thor was hitting a match from the second index which occurred on an earlier node.

Confirmed that is what's happening.
Thor sees a match in the 2nd index 1st (on an earlier node).

Reply:
So Thor is interleaving. In this case the width the Thor is the same as the indexes.
So the 1st slave, is processing the 1st parts of the index.
The 2nd slave is processing the 2nd parts, etc.

Thor sees a match from the 2nd index first, the 1st index has matches, but they come in partitions read on later nodes, so are not part of the i[1].doc result.

Which I think is correct.

Reply:
Btw, hthor result is same as roxie. So it looks like both read all parts of subfile1, before reading subfile2 ..